jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -810,24 +810,50 @@ YUI.add('juju-env-api', function(Y) {
         this.setConnectedAttr('maasServer', null);
         return;
       }
-      this.modelGet(data => {
-        if (data.err) {
-          console.warn('error calling ModelGet API: ' + data.err);
+      this.cloudInfo(data.cloud, data => {
+        if (!data) {
+          console.log('cannot retrieve maas-server from Cloud call');
+          this.setConnectedAttr('maasServer', null);
           return;
         }
-        const maasServer = data.config['maas-server'];
+        if (data.error) {
+          console.warn('error calling Cloud API: ' + data.error);
```

Comment:
Since this seems to come from response.results[0], the error field here is an object with a message attribute.
But, as suggested above, we can change this so that we always return an error string, as in listKeys.